apps/wallet/src/ui/app/pages/home/nfts/index.tsx

```diff
@@ -1,26 +1,76 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useOnScreen, useGetOwnedObjects } from '@mysten/core';
 import { getObjectDisplay, type SuiObjectData } from '@mysten/sui.js';
+import { useRef, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 
 import { useActiveAddress } from '_app/hooks/useActiveAddress';
 import Alert from '_components/alert';
 import { ErrorBoundary } from '_components/error-boundary';
 import Loading from '_components/loading';
+import LoadingSpinner from '_components/loading/LoadingIndicator';
 import { NFTDisplayCard } from '_components/nft-display';
-import { useObjectsOwnedByAddress } from '_hooks';
 import PageTitle from '_src/ui/app/shared/PageTitle';
 
+// Set the default limit to 10
+const DEFAULT_NFT_LIMIT = 10;
+const MAX_FETCH_LIMIT = 50;
+
 function NftsPage() {
     const accountAddress = useActiveAddress();
-    const { data, isLoading, error, isError } = useObjectsOwnedByAddress(
+    const {
+        data,
+        isLoading,
+        error,
+        isError,
+        isFetchingNextPage,
+        hasNextPage,
+        fetchNextPage,
+        isInitialLoading,
+    } = useGetOwnedObjects(
         accountAddress,
-        { options: { showType: true, showDisplay: true } }
+        {
+            MatchNone: [{ StructType: '0x2::coin::Coin' }],
+        },
+        MAX_FETCH_LIMIT
     );
-    const nfts = data?.data
-        ?.filter((resp) => !!getObjectDisplay(resp).data)
-        .map(({ data }) => data as SuiObjectData);
+    const observerElem = useRef<HTMLDivElement | null>(null);
+    const { isIntersecting } = useOnScreen(observerElem);
+    const isSpinnerVisible = isFetchingNextPage && hasNextPage;
+
+    const nfts =
+        data?.pages
+            .flatMap((page) => page.data)
+            .filter((resp) => !!getObjectDisplay(resp).data)
+            .map(({ data }) => data as SuiObjectData) || [];
+
+    useEffect(() => {
+        if (
+            (isIntersecting && hasNextPage && !isFetchingNextPage) ||
+            (nfts.length < DEFAULT_NFT_LIMIT &&
```

Comment:
I probably wouldn't do this special case. The onscreen observer should handle the case where there's not enough to populate one page. I think that's enough.